apps/angular/di/src/app/currency-setter.directive.ts

```diff
@@ -0,0 +1,18 @@
+import { Directive, effect, inject, input } from '@angular/core';
+import { CurrencyService } from './currency.service';
+
+@Directive({
+  selector: '[currencySetter]',
+  providers: [CurrencyService],
+  standalone: true,
+})
+export class CurrencySetterDirective {
+  currencySetter = input.required<string>();
+  #currencyService = inject(CurrencyService);
+
+  constructor() {
+    effect(() => {
+      this.#currencyService.patchState({ code: this.currencySetter() });
```

Comment:
this will cause an error in console, since signal effects aren't allowed to set signals (and component store is signal based now)